### DISCUSSION TITLE
Transform Message or Hook for 0.4

### DISCUSSION THREAD

**Discussion by akurniawan** [microsoft/autogen#4668] [General]
Hi, I have been playing around with the new 0.4 framework but I couldn't see something similar to [transform message](https://microsoft.github.io/autogen/0.2/docs/topics/handling_long_contexts/intro_to_transform_messages/) for modifying incoming message to the agent with a template or a hook that I can use to modify llm's output. The idea is to use this along with groupchat. Is there something like this implemented already and I missed it? Thanks!

**Comment by ekzhu** [comment]
Thanks for giving it a try! We are working on the memory module #4438. It is still WIP. The general idea is that this will be used to provide additional context or transform the context.

**Comment by akurniawan** [comment]
Thanks for your reply! I tried to look into the PR, it seems that's intended for adding extra context like RAG. My use case is a bit different, each agents will have it's own message format. For example, instead of just passing QUESTION into the model directly, I want to reformat it into something like:

QUESTION:
{original question}

and for another agent, the template would be completely different.

From the PR it seems like it's just adding additional context prior to the original message and keep the original QUESTION intact.

This kind of templating would be difficult to do in GroupChat since .send_message is wrapped inside .run_stream for all agents.

**Comment by akurniawan** [comment]
Sorry, didn't read the conversation thoroughly, seems like you have already addressed it with your comment. Will monitor the progress closely. Thanks!

**Reply by ekzhu** [reply]
You are welcome to contribute to the discussion. Also we are on discord: https://aka.ms/autogen-discord
